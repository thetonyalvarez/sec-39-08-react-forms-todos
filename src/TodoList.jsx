import { useState } from 'react'
import { v4 as uuid } from 'uuid';
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'
import Box from '@mui/material/Box'
import Container from '@mui/material/Container'
import Grid from '@mui/material/Grid'
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';

import './TodoList.css'



const TodoList = () => {
  const [todos, setTodos] = useState([])
  
  // add new todo
  const addTodo = (todo) => {
    let newTodo = {...todo, id: uuid()}
    setTodos(todos => [...todos, newTodo])
  }
  // end addTodo

  // remove todo
  const removeTodo = (id) => {
    setTodos(todos => todos.filter(todo => todo.id !== id))
  }
  // end todo

  // render all todos
  const renderTodos = () => {
    return (
      <Grid item xs={12} md={6} data-testid="TodoList">
        <Table className="todoGrid" data-testid="Table">
          <TableBody>
            {todos.map(todo => (
              <Todo
                key={todo.id}
                id={todo.id}
                task={todo.task}
                handleRemove={removeTodo}
                data-testid="Todo"
              />
            ))}
          </TableBody>
        </Table>
      </Grid>
    )
  }

  return (
    <Grid container spacing={8} className="TodoList">
      <NewTodoForm addTodo={addTodo}/>
      {renderTodos()}
    </Grid>
  )
}

export default TodoList;