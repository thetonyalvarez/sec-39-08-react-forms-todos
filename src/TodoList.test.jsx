import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import TodoList from './TodoList'

afterEach(cleanup)

// smoke test
it("renders without crashing", function() {
  render(<TodoList />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("shows empty TodoList (aka no Todo item) on start", function() {
  const { getByTestId } = render(<TodoList />);
  expect(getByTestId('TodoList')).not.toContainHTML("class='Todo'")
});