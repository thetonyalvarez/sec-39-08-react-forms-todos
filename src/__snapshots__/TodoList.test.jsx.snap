// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 TodoList css-r1hick-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
    >
      <form
        autocomplete="off"
        class="css-ow1s07-MuiStack-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1sqs9ir-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="task"
            id="task-label"
          >
            Task
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="task"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Add new task
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      data-testid="TodoList"
    >
      <table
        class="MuiTable-root todoGrid css-rqglhn-MuiTable-root"
        data-testid="Table"
      >
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
